integration-auth/src/test/java/org/symphonyoss/integration/authentication/AuthenticationProxyImplTest.java

```diff
@@ -124,6 +132,37 @@ public void setup() {
     kmToken2.setToken(KM_TOKEN2);
   }
 
+  @Test(expected = UnregisteredUserAuthException.class)
+  public void testUnregisteredUserAuthException() throws RemoteApiException {
+    Whitebox.setInternalState(proxy, "authContexts", new ConcurrentHashMap<>());
```

Comment:
You don't need to change the field 'authContexts' via reflection. You should call the 'authenticate' method using a different userId.

Something like that: proxy.authenticate("test");